taxonomy/utils.py

```diff
@@ -144,6 +190,72 @@ def refresh_course_skills(courses, should_commit_to_db):
     )
 
 
+def refresh_program_skills(programs, should_commit_to_db):
+    """
+    Refresh the skills associated with the provided programs.
+    """
+    all_failures = []
+    success_programs_count = 0
+    skipped_programs_count = 0
+
+    for index, program in enumerate(programs, start=1):
+        program_overview = program['overview']
+        if program_overview:
+            try:
+                program_skills = call_emsi_api(program_overview, index)
+            except TaxonomyAPIError:
+                message = f'[TAXONOMY] API Error for program uuid: {program["uuid"]}'
+                LOGGER.error(message)
+                all_failures.append((program['uuid'], message))
+                continue
+
+            try:
+                failures = process_program_skills_data(program, program_skills, should_commit_to_db)
+                if failures:
+                    LOGGER.info('[TAXONOMY] Skills data received from EMSI. Skills: [%s]', program_skills)
+                    all_failures += failures
+                else:
+                    success_programs_count += 1
+            except Exception as ex:  # pylint: disable=broad-except
+                LOGGER.info('[TAXONOMY] Skills data received from EMSI. Skills: [%s]', program_skills)
+                message = f'[TAXONOMY] Exception for program uuid: {program["uuid"]} Error: {ex}'
+                LOGGER.error(message)
+                all_failures.append((program["uuid"], message))
+        else:
+            skipped_programs_count += 1
+
+    LOGGER.info(
+        '[TAXONOMY] Refresh program skills process completed. \n'
+        'Failures: %s \n'
+        'Total Programs Updated Successfully: %s \n'
+        'Total Programs Skipped: %s \n'
+        'Total Failures: %s \n',
+        all_failures,
+        success_programs_count,
+        skipped_programs_count,
+        len(all_failures),
```

Comment:
please be consistent with string formatting method. You have used f-string method above.